src/entity.rs

```diff
@@ -1619,10 +1726,267 @@ impl Entity {
             pairs.push(CodePair::new_string(5, &handle.as_string()));
         }
     }
+
+    fn push_boundary_edge(pairs: &mut Vec<CodePair>, edge: &BoundaryPathEdge) {
+        if let crate::BoundaryPathEdge::Polyline { vertex } = edge {
+            pairs.push(CodePair::new_f64(10, vertex.x));
+            pairs.push(CodePair::new_f64(20, vertex.y));
+            pairs.push(CodePair::new_f64(42, vertex.z));
```

Comment:
Code `42` is a bulge value, not a Z-value of the vertex.  Check the section "Polyline boundary data group codes" on [this page](https://help.autodesk.com/view/OARX/2020/ENU/?guid=GUID-DC5215D6-E73F-4DFF-8BE9-01CA9610FAEE).